service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -112,13 +112,18 @@ public ResponseEntity<ApiCreatedWorkspace> createWorkspace(
     Optional<SpendProfileId> spendProfileId =
         Optional.ofNullable(body.getSpendProfile()).map(SpendProfileId::new);
 
+    // ET uses userFacingId; CWB doesn't. Schema enforces that userFacingId must be set. CWB doesn't pass
+    // userFacingId in request, so use id. Prefix with "a" because userFacingId must start with letter.
+    String userFacingId = body.getUserFacingId() != null ? body.getUserFacingId() : "a" + body.getId();
```

Comment:
You can also do this using Optional like
```
String userFacingId = Optional.ofNullable(body.getUserFacingId()).orElse("a" + body.getId());
```
Not that you have to. I am not afraid of nulls, but there seems to be a growing preference for using Optional for this case.